server/src/main/java/org/solarbank/server/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,60 @@
+package org.solarbank.server;
+
+import javax.money.UnknownCurrencyException;
+import org.solarbank.server.Messages;
+import org.solarbank.server.dto.ErrorResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ExceptionHandler
+    @ResponseStatus
+    public ResponseEntity<ErrorResponse> handleUnknownCurrencyException(UnknownCurrencyException ex) {
+        ErrorResponse errorResponse = new ErrorResponse(
+                HttpStatus.BAD_REQUEST.value(),
+                HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                ex.getMessage()
+        );
+        System.out.println(errorResponse);
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler
+    @ResponseStatus
+    public ResponseEntity<ErrorResponse> DefaultExceptionHandler(Exception ex) {
+        ErrorResponse errorResponse = new ErrorResponse(
+                HttpStatus.INTERNAL_SERVER_ERROR.value(),
+                HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase(),
+                Messages.INTERNAL_SERVER_ERROR.getMessage()
+        );
+        System.out.println(errorResponse);
+        return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler
+    @ResponseStatus
+    public ResponseEntity<ErrorResponse> handleValidationException(
```

Comment:
You had a comment about this, but not sure what I need to do.

Reply:
You need to check to see if  `.getBindingResult()` or `getFieldErrors()` can return any exceptions. If so catch those exceptions and decide how to handle them. 

I would guess the most logical thing to do would be to append a fallback error message to your `errorMessage` object in the case that there is some issue with retrieving the validation errors from the exception or `getFieldErrors` returns an empty list.

Otherwise if an uncaught exception is thrown a 500 will go back to the client or if `getFieldErrors` is empty an empty message will be returned with the `errorResponse`.

In all cases you need to write unit tests to cover whatever you decide to implement.

Reply:
I've added in some logic to handle any exceptions or null empty field errors. Added a test for the missing filed errors. I'm unable to test null BindingResult as throwing an MethodArgumentNotValidException with a null binding result triggers an IllegalArgumentException: java.lang.IllegalArgumentException: BindingResult must not be null.

Reply:
added logic for handling exceptions and added tests for all conditions